docs/large_project_structure.md

```diff
@@ -0,0 +1,107 @@
+If you have multiple a large project with maybe multiple `streams` we recommend the following project structure:
+
+```
+├── my-project
+│   ├── my_project
+│   │   ├── __init__.py
+│   │   ├── app.py
+│   │   ├── resources.py
+│   │   └── streams.py
+│   │── pyproject.toml
+│   │── README.md
+```
+
+- The file `my_project/resouces.py` contains the creation of the `StreamEngine`
+- The file `my_project/app.py` contains the entrypoint of your program
+- The file `my_project/streams.py` contains all the `Streams`
+
+A full project example ready to use can be found [here](https://github.com/kpn/kstreams/tree/master/examples/recommended-worker-app)
+
+!!! note
+    This is just a recommendation, there are many ways to structure your project
+
+## Resources
+
+This python module contains any global resource that will be used later in the application, for example `DB connections` or the `StreamEngine`. Typically we will have the following:
+
+```python
+import kstreams
+
+stream_engine = kstreams.create_engine(title="my-stream-engine")
+```
+
+Then later `stream_engine` can be resused to start the application
+
+## Streams
+
+When starting your project you can have `N` number of `Streams`, let's say in `streams.py` module. All of the `Streams` will run next to each other and because they are in the same project it is really easy to share common code. However, this comes with a downside of `scalability` as it is not possible to take the advantages of `kafka` and scale up `Streams` individually. In next versions the `StreamEngine` will be able to select which `Stream/s` should run to mitigate this issue. Typically, your `streams.py` will look like:
+
+```python
+import logging
+
+from kstreams import ConsumerRecord, Send, stream
+
+logger = logging.getLogger(__name__)
+
+
+@stream("local--hello-world", group_id="example-group")
```

Comment:
I would even recommend building the `Stream` on the `app.py`. So you can grab this function, which contains the business logic, on `ipython` and try it. 

Maybe in this case with the consumer record you have to take extra steps to build the `ConsumerRecord`. But in the future if we support passing the `Value` with a pydantic model, then it becomes super straightforward.
What do you think?

Reply:
Ah, do you mean something like this:

```python
# stream_handlers.py

async def my_stream_handler(cr: ConsumerRecord, send: Send) -> None:
    logger.info(f"showing bytes: {cr.value}")
    value = f"Event confirmed. {cr.value}"

    await send(
        "another-topic-to-wink",
        value=value.encode(),
        key="1",
    )
```

and 

```python
# streams.py
from kstreams import Stream

from .stream_handlers import my_stream_handler(

my_stream = Stream(
   ...,
   func=my_stream_handler(,
)
```

If this is the case we should also change the [recommended-worker-app](https://github.com/kpn/kstreams/tree/master/examples/recommended-worker-app)